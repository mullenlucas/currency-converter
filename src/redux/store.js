// import { configureStore } from '@reduxjs/toolkit';

// const store = configureStore({
//   reducer: {
//     convert: convertReducer,
//   },
// });

// export default store;
