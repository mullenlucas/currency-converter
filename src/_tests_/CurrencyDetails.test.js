import React from 'react';
import { render } from '@testing-library/react';
import { BrowserRouter as Router, Route, Routes } from 'react-router-dom';
import { Provider } from 'react-redux';
import { createBrowserHistory } from 'history';
import CurrencyDetails from '../displays/detailsPage';
import store from '../redux/store';

describe('Currency details test', () => {
  test('Currency details snapshot rendering...', () => {
    const history = createBrowserHistory();

    const curr = {
      id: 94,
      name: 'Argentine Peso',
      acronym: 'ARS',
      icon: 'https://currencyfreaks.com/photos/flags/ars.png',
      status: 'AVAILABLE',
      availableFrom: '1992, 1, 2',
      availableTill: '2022-10-14',
      countryCode: 'AR',
      countryName: 'Argentina',
      USDconversion: '151.25',
    };

    history.push('/details', { curr });
    const tree = render(
      <React.StrictMode>
        <Provider store={store}>
          <Router history={history}>
            <Routes>
              <Route path="/details" element={<CurrencyDetails />} />
            </Routes>
          </Router>
        </Provider>
      </React.StrictMode>,
    );
    expect(tree).toMatchSnapshot();
  });
});
