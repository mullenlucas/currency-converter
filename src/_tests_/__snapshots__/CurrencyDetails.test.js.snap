// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Currency details test Currency details snapshot rendering... 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="currency-detail"
      >
        <div
          class="curr-detail-card"
        >
          <div
            class="curr-detail-header"
          >
            <h2>
              Argentine Peso
            </h2>
            <h2>
              ARS
            </h2>
          </div>
          <img
            alt="currency-icon"
            class="currency-icon-detail"
            src="https://currencyfreaks.com/photos/flags/ars.png"
          />
          <div
            class="curr-detail-footer"
          >
            <h3>
              ARS
               
              per U$D unit:
            </h3>
            <h3>
              151.25
            </h3>
          </div>
        </div>
        <div
          class="currency-info"
        >
          <h3>
            Detailed Information
          </h3>
          <div
            class="info-cards"
          >
            <div
              class="detail-card typeA"
            >
              <img
                alt="success-logo"
                class="info-logo"
                src="success-icon.png"
              />
              <div
                class="li"
              >
                <h4>
                  Status
                </h4>
                <p>
                  AVAILABLE
                </p>
              </div>
            </div>
            <div
              class="detail-card typeB"
            >
              <img
                alt="world-logo"
                class="info-logo"
                src="world-icon.png"
              />
              <div
                class="li"
              >
                <h4>
                  Country
                </h4>
                <p>
                  Argentina
                  (
                  AR
                  )
                </p>
              </div>
            </div>
            <div
              class="detail-card typeB"
            >
              <img
                alt="start-logo"
                class="info-logo"
                src="start-icon.png"
              />
              <div
                class="li"
              >
                <h4>
                  Available Since
                </h4>
                <p>
                  1992, 1, 2
                </p>
              </div>
            </div>
            <div
              class="detail-card typeA"
            >
              <img
                alt="clock-logo"
                class="info-logo"
                src="clock-icon.png"
              />
              <div
                class="li"
              >
                <h4>
                  Available Until
                </h4>
                <p>
                  2022-10-14
                </p>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="currency-detail"
    >
      <div
        class="curr-detail-card"
      >
        <div
          class="curr-detail-header"
        >
          <h2>
            Argentine Peso
          </h2>
          <h2>
            ARS
          </h2>
        </div>
        <img
          alt="currency-icon"
          class="currency-icon-detail"
          src="https://currencyfreaks.com/photos/flags/ars.png"
        />
        <div
          class="curr-detail-footer"
        >
          <h3>
            ARS
             
            per U$D unit:
          </h3>
          <h3>
            151.25
          </h3>
        </div>
      </div>
      <div
        class="currency-info"
      >
        <h3>
          Detailed Information
        </h3>
        <div
          class="info-cards"
        >
          <div
            class="detail-card typeA"
          >
            <img
              alt="success-logo"
              class="info-logo"
              src="success-icon.png"
            />
            <div
              class="li"
            >
              <h4>
                Status
              </h4>
              <p>
                AVAILABLE
              </p>
            </div>
          </div>
          <div
            class="detail-card typeB"
          >
            <img
              alt="world-logo"
              class="info-logo"
              src="world-icon.png"
            />
            <div
              class="li"
            >
              <h4>
                Country
              </h4>
              <p>
                Argentina
                (
                AR
                )
              </p>
            </div>
          </div>
          <div
            class="detail-card typeB"
          >
            <img
              alt="start-logo"
              class="info-logo"
              src="start-icon.png"
            />
            <div
              class="li"
            >
              <h4>
                Available Since
              </h4>
              <p>
                1992, 1, 2
              </p>
            </div>
          </div>
          <div
            class="detail-card typeA"
          >
            <img
              alt="clock-logo"
              class="info-logo"
              src="clock-icon.png"
            />
            <div
              class="li"
            >
              <h4>
                Available Until
              </h4>
              <p>
                2022-10-14
              </p>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
